activerecord/lib/active_record/database_configurations/database_config.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  class DatabaseConfigurations
+    class DatabaseConfig # :nodoc:
```

Comment:
In the `DatabaseConfigurations` documentation we say we return `DatabaseConfig` but we mark `DatabaseConfig` as private API. Do we need to make it public given we are exposing this object? Or, we don't want people to be dealing with this object?

Reply:
I updated this - so the `DatabaseConfig` object is just a parent class but I clarified that in docs, and then documented `HashConfig` and `UrlConfig`.